# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1207759254 14400
#      Wed Apr 09 12:40:54 2008 -0400
# Branch release-3-0-x
# Node ID c7070f6cadc8d577f45a79cf00bfb65da33f25b4
# Parent  9f83fb63f764e8706eee7315fec001a8e835f42e
Don't allow the use of lists in assert.m

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-04-09  David Bateman  <dbateman@free.fr>
+
+	* testfun/assert.m: Don't allow cond and expected to be lists.
+
 2008-04-04  David Bateman  <dbateman@free.fr>
 
 	* deprecated/beta_cdf.m deprecated/beta_inv.m
 	deprecated/beta_pdf.m deprecated/beta_rnd.m
 	deprecated/binomial_cdf.m deprecated/binomial_inv.m
 	deprecated/binomial_pdf.m deprecated/binomial_rnd.m
 	deprecated/chisquare_cdf.m deprecated/chisquare_inv.m
 	deprecated/chisquare_pdf.m deprecated/chisquare_rnd.m
diff --git a/scripts/testfun/assert.m b/scripts/testfun/assert.m
--- a/scripts/testfun/assert.m
+++ b/scripts/testfun/assert.m
@@ -72,29 +72,16 @@ function assert (cond, expected, tol)
 
   coda = "";
   iserror = 0;
   if (nargin == 1)
     if (! isnumeric (cond) || ! all (cond(:)))
       error ("assert %s failed", in); # say which elements failed?
     endif
   
-  elseif (is_list (cond))
-    if (! is_list (expected) || length (cond) != length (expected))
-      iserror = 1;
-    else
-      try
-	for i = 1:length (cond)
-	  assert (nth (cond, i), nth (expected, i));
-	endfor
-      catch
-	iserror = 1;
-      end_try_catch
-    endif
-
   elseif (ischar (expected))
     iserror = (! ischar (cond) || ! strcmp (cond, expected));
 
   elseif (iscell (expected))
     if (! iscell (cond) || any (size (cond) != size (expected)))
       iserror = 1;
     else
       try
